build.gradle

```diff
@@ -893,6 +895,41 @@ subprojects {
                         }
                     }
                     projectMakePatches.dependsOn "project${child.name}MakePatches"
+
+
+                    if (rootProject.ext.COMPILERS_JSON != null) {
+                        def json = rootProject.ext.COMPILERS_JSON
+                        task "testRecompile${child.name}"
+                        testRecompileAll.dependsOn("testRecompile${child.name}")
+                        json.each {
+                            def jvmName = it.key
+                            def jvmData = it.value
+                            task "clearTestRecompile${child.name}_${jvmName}"(type: Delete) {
+                                delete new File(dir, "build/testOutput_${child.name}_${jvmName}")
+                            }
+                            task "testRecompile${child.name}_${jvmName}"(type: JavaCompile) {
+                                dependsOn("project${child.name}ApplyPatches", "clearTestRecompile${child.name}_${jvmName}")
+                                options.fork = true
+                                options.compilerArgs.addAll(jvmData.extraCompilerArgs)
+                                options.forkOptions.with {
+                                    executable = jvmData.compilerExec
+                                    jvmArgs = jvmData.compilerJvmArgs
+                                }
+                                destinationDir = new File(dir, "build/testOutput_${child.name}_${jvmName}")
+                                source = new File(dir, "src/main/java")
+                                sourceCompatibility = '1.8'
+                                targetCompatibility = '1.8'
```

Comment:
I also just realized that it may be useful to have this be specified per MC version. Or is that something that is going to be changed when the time comes MC actually needs something newer than java 8?

Reply:
Isnt this exposed in the config.json? if not we should.

Reply:
Doesn't seem to be there. Should I add that in this PR?

Reply:
That would need a fg end thing.. I'll add it to my local fork which already has config v2

Reply:
But if you wanna setup support over for that then sure

Reply:
In that case, I think would be better to do that with the config v2, especially as it seems to be a relatively simple change.